app/channels/application_cable/connection.rb

```diff
@@ -9,10 +9,26 @@ def connect
 
     private
 
+    # rubocop:disable Metrics/PerceivedComplexity
     def find_verified_agent
       session = cookies.encrypted[Rails.application.config.session_options[:key]]
-      agent_id_from_session = session["warden.user.agent.key"].first.first
-      Agent.find_by(id: agent_id_from_session) || reject_unauthorized_connection
+      reject_unauthorized_connection and return unless session
+
+      agent_id_from_session = session["warden.user.agent.key"]&.first&.first
+
+      # Débugging temporaire
+      unless agent_id_from_session
+        Rails.logger.debug { "Warn: Cookie de session trouvé mais sans ID. Session: #{session.inspect}" }
+      end
+
+      if agent_id_from_session
+        Agent.find_by(id: agent_id_from_session) || reject_unauthorized_connection
+      else
+        # On est par exemple dans ce cas lorsqu'un agent laisse son navigateur
+        # ouvert toute la nuit et que le cookie a expiré le lendemain.
+        reject_unauthorized_connection
+      end
```

Comment:
je sais pas si c’est mieux mais :

```suggestion
      if agent_id_from_session.present?
        agent = Agent.find_by(id: agent_id_from_session) 
        return agent if agent.present?
      else
        # On est par exemple dans ce cas lorsqu'un agent laisse son navigateur
        # ouvert toute la nuit et que le cookie a expiré le lendemain.
        Rails.logger.debug { "Warn: Cookie de session trouvé mais sans ID. Session: #{session.inspect}" }
      end
           
      reject_unauthorized_connection
```

Reply:
Ça me plait, j'ai fait un truc qui ressemble !